scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt

```diff
@@ -114,27 +121,20 @@ internal fun <T : Summarizable> List<T>.mapSummary(
  */
 internal fun mapSnippetFindings(
     rawResults: RawResults,
-    scannerConfig: ScannerConfiguration,
     issues: MutableList<Issue>
 ): Set<SnippetFinding> {
-    val fakeLocation = TextLocation(".", TextLocation.UNKNOWN_LINE)
-
     return rawResults.listSnippets.flatMap { (file, rawSnippets) ->
         rawSnippets.map {
-            val license = it.artifactLicense?.let {
-                runCatching {
-                    LicenseFinding.createAndMap(
-                        it,
-                        fakeLocation,
-                        detectedLicenseMapping = scannerConfig.detectedLicenseMapping
-                    ).license
-                }.onFailure { spdxException ->
+            val license = it.artifactLicense?.let { artifactLicense ->
```

Comment:
After rebasing (which introduces the `snippet` lambda variable) I'd propose the slightly more compact:

            val license = snippet.artifactLicense?.let {
                DeclaredLicenseProcessor.process(it).also { expression ->
                    if (expression == null) {
                        issues += FossId.createAndLogIssue(
                            source = "FossId",
                            message = "Failed to map license '$it' as an SPDX expression."
                        )
                    }
                }
            } ?: SpdxConstants.NOASSERTION.toSpdx()